react-app/src/pages/RecursivePage.tsx

```diff
@@ -20,7 +26,7 @@ export default function RecursivePage() {
                         placeholder="Amount of components"
                         min="0"
                     />
-                    <button onClick={() => setMaxDepth(inputValue)}>Generate Components</button>
+                    <button onClick={() => handleGenerateComponents()}>Generate Components</button>
```

Comment:
This could be a pointer to handleGenerateComponents istead of lambda function

Reply:
fixed